### PR TITLE
Add generate genesis script

### PR DESCRIPTION
**Pull request by anilcse**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,3 +8,5 @@ prelaunch-2:
 regen-1:
 	go run . build-genesis regen-1
 	mv -f regen-1/genesis.json regen-1/genesis-prelaunch.json
+	go run . build-genesis regen-prelaunch-1 --errors-as-warnings
```
Comment (clevinson) [inline Makefile]: We shouldn't need to run `regen-prelaunch-1` here, just running the above line (line 9) should be sufficient.

File: scripts/gen-genesis.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+cd regen-ledger
+git checkout v1.0.0-rc3
+make build
+chmod +x ./build/regen
+
+./build/regen init --chain-id $CHAIN_ID validator --home $REGEN_HOME
+
+echo "..........Fetching genesis......."
+rm -rf $REGEN_HOME/config/genesis.json
+curl -s https://raw.githubusercontent.com/regen-network/mainnet/main/$CHAIN_ID/genesis-prelaunch.json >$REGEN_HOME/config/genesis.json
```
Comment (clevinson) [inline scripts/gen-genesis.sh]: I think this would be cleaner if it pulls from `../regen-1/genesis-prelaunch.json` in the current directory / repo, as opposed to pulling from github again.

**Comment by anilcse** [comment]
@clevinson Chainflow's GenTx is invalid. They missed the memo field. They also missed the deadline so Github action also didn't catch it (It was set to validate the PRs which comes before the deadline). Hope this is fine, they can join post genesis

**Comment by clevinson** [comment]
Looks good. Thanks @anilCSE ! Let's  hold on merging until #116 is in to master so we can add final genesis & checksum to this PR.

File: regen-1/gentxs/gentx-b3929bf6504c5a87955d171158d5c338737d7e98.json [hunk 3]
```diff
@@ -1 +0,0 @@
...":[{"@type":"/cosmos.staking.v1beta1.MsgCreateValidator","description":{"moniker":"Chainflow","identity":"","website":"","security_contact":"","details":"Independent Validator Providing Fair and Equitable Staking for All #KeepStakeDecentralized https://chainflow.io/staking"},"commission":{"rate":"0.080000000000000000","max_rate":"1.000000000000000000","max_change_rate":"0.050000000000000000"},"min_self_delegation":"1","delegator_address":"regen1snn4uhxh04gzpgk4l8naw3n6fu7ucwx3zfv0hn","validator_address":"regenvaloper1snn4uhxh04gzpgk4l8naw3n6fu7ucwx37w320z","pubkey":{"@type":"/cosmos.crypto.ed25519.PubKey","key":"4ZO9Oj2esqGYQA6KiND6MYa8X/TCq3Xc4zFi1wytSHE="},"value":{"denom":"uregen","amount":"1000000000"}}],"memo":"","timeout_height":"0","extension_options":[],"non_critical_extension_options":[]},"auth_info":{"signer_infos":[{"public_key":{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AibuN5foIxPrciwIokFyGyAX3BGkUY6La/XEVUTU3Pcf"},"mode_info":{"single":{"mode":"SIGN_MODE_LEGACY_AMINO_JSON"}},"sequence":"0"}],"fee":{"amount":[],"gas_limit":"200000","payer":"","granter":""}},"signatures":["MFNDKzTe9wXI+s8ZR8XornsnvNYFi7F8vY8m8klkXmY87I8PTCAmdBEnL3AvMEH593PEUY909xbsOhhBqUp1lg=="]}
```
Comment (aaronc) [inline regen-1/gentxs/gentx-b3929bf6504c5a87955d171158d5c338737d7e98.json:1]: Why was this deleted?

File: Makefile [hunk 4]
```diff
@@ -8,3 +8,4 @@ prelaunch-3:
 regen-1:
 	go run . build-genesis regen-1
 	mv -f regen-1/genesis.json regen-1/genesis-prelaunch.json
```
Comment (aaronc) [inline Makefile:10]: Why not just use genesis.json now instead of genesis-prelaunch?

**Comment by aaronc** [comment]
> @clevinson Chainflow's GenTx is invalid. They missed the memo field. They also missed the deadline so Github action also didn't catch it (It was set to validate the PRs which comes before the deadline). Hope this is fine, they can join post genesis

Have you let them know? I guess it's too late now?

File: regen-1/gentxs/gentx-b3929bf6504c5a87955d171158d5c338737d7e98.json (hunk 3 above)
Comment (aaronc) [reply regen-1/gentxs/gentx-b3929bf6504c5a87955d171158d5c338737d7e98.json:1]: Nevermind I see

**Comment by clevinson** [comment]
@aaronc I've alerted @chris-remus on discord. I think at this point we shouldn't be waiting for anyone. Its unfortunate, but I think its important that we get the final published genesis out by EOD today for community review.

File: regen-1/gentxs/gentx-b3929bf6504c5a87955d171158d5c338737d7e98.json (hunk 3 above)
Comment (anilCSE) [reply regen-1/gentxs/gentx-b3929bf6504c5a87955d171158d5c338737d7e98.json:1]: @aaronc the gentx is invalid: https://github.com/regen-network/mainnet/pull/114#issuecomment-817932383

File: Makefile (hunk 4 above)
Comment (anilcse) [reply Makefile:10]: We'll use `genesis.json` for final genesis with gentxs. So, let's use `genesis-prelaunch`

Comment (clevinson) [reply Makefile:10]: I agree that it'd be nice to not have `genesis-prelaunch.json` checked in to the repo now that we have the final `genesis.json`. IMO either we should remove `genesis-prelaunch.json` as a last step in this makefile, or just drop this `mv -f regen-1/genesis.json regen-1/genesis-prelaunch.json` line, and then overwrite `genesis.json` in the `gen-genesis.sh` script.

Comment (aaronc) [reply Makefile:10]: Up to you guys

**Comment by anilcse** [comment]
@clevinson @aaronc I have updated the PR with final genesis and checksum. Can you also run `make regen-1` from this branch and confirm there are no git changes.

**Comment by clevinson** [comment]
`make regen-1` working for me. no git diff. Thanks @anilCSE !

**Review by clevinson — APPROVED** [review]

**Review by aaronc — APPROVED** [review]
ACK 🎉
